Mathlib/Data/List/Basic.lean

```diff
@@ -485,6 +490,9 @@ theorem replicate_left_injective (a : α) : Injective (replicate · a) :=
   (replicate_left_injective a).eq_iff
 #align list.replicate_left_inj List.replicate_left_inj
 
+@[simp] theorem head_replicate {n} {a : α} (h : List.replicate n a ≠ []) :
+  head _ h = a := by
```

Comment:
I would put the conclusion on the same line.
Please put separate tactic calls on different lines: altogether, I'd write this like
```lean
@[simp]
theorem head_replicate {n} {a : α} (h : List.replicate n a ≠ []) : head _ h = a := by
  cases n <;> simp at *
  exfalso
  exact h rfl